tests/lib/rules/counter-badge-needs-count.js

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require("../../../lib/rules/counter-badge-needs-count"),
+    RuleTester = require("eslint").RuleTester;
+
+RuleTester.setDefaultConfig({
+    parserOptions: {
+        ecmaVersion: 6,
+        ecmaFeatures: {
+            jsx: true
+        }
+    }
+});
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester();
+ruleTester.run("counter-badge-needs-count", rule, {
+    valid: [
+        `<CounterBadge count={5} appearance="filled" />`,
+        `<CounterBadge icon={<PasteIcon aria-label="paste" />} count={1} />`,
```

Comment:
The CounterBadge component applies aria-hidden to the icon, so it's only for looks and we don't need an aria-label if the count prop is provided.